### PR TITLE
Mentorship page changes, Course page changes, Admin View User page changes

### PR DESCRIPTION
**Pull request by rawrrencee**
# Changelog:

1. Created a PageHeader component that is similar to AntD's one but more flexible and way more responsive. 
2. Zhng View Purchased Course As Student page
3. Shrink Ongoing Mentorships card + Add Sensei image to it
4. Add display Reviews with Pagination to courses and mentorships
5. Add common getImage utils function to return image..Can use if want, no need to refactor (too much lol)
6. Update formatTime function to not return seconds in the time (not necessary)
7. Update Social Follow Button to disable itself when you're looking at yourself / you are an ADMIN
8. add back search function for admin and sensei, but sensei can only search for users
9. Merge Admin View Sensei and View Student into a single page @y1bs - I needed to do this because the "View a profile" url from the search result doesn't contain User Type, so I couldn't send the admin to the right page after searching...So I just merged the 2 pages tgt.
10. All urls to send to sensei/student as admin have been changed to the new endpoint
11. jwt functions not touched, worried for dependency checks (too tired to check properly)
10. Make admin view course page the same as the student page but use conditions to hide elements
11. Remove old admin page (verify sensei)
12. Update notifications text to remove 'course **request**'
13. Zhng Mentorship Contract page for student
14. Hide refund button on contract page if no mentorship passes
15. Add Paragraph to mentorship contracts table
16. Refactor Reviews Modal into Common/Reviews

# Merge/Review Order

These other PRs should be merged or reviewed first

This PR blocks these other PRs

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

### Course as public
image
image

### Courses as Student
image
image

### Ongoing Mentorships card
image

### Search
image

### Mentorship Contract
image
image
image
image
image

### Mentorship Contracts Table
image

### CODE REVIEW

**Review by y1bs — APPROVED** [review]
Looks ok

File: src/router.js [hunk 1]
```diff
@@ -192,12 +192,12 @@ const routes = [
...
     path: '/admin/course-content-management/:id',
-    Component: lazy(() => import('pages/admin/courses/view')),
+    Component: lazy(() => import('pages/student/dashboard/courses/view')),
     exact: true,
   },
   {
     path: '/admin/course-content-management/:courseId/view-lesson/:lessonId',
-    Component: lazy(() => import('pages/admin/courses/view-lesson')),
+    Component: lazy(() => import('pages/student/dashboard/courses/view-lesson')),
     exact: true,
```
Comment (natalieloke) [inline src/router.js:196]: for next time: if this page is used for admin also then maybe we should put it in pages/courses instead

File: src/services/admin/index.js [hunk 2]
```diff
@@ -161,6 +161,18 @@ export async function getAllSenseis() {
...
+    .get(`/user/${accountId}`, { withCredentials: true })
+    .then(response => {
+      if (response && !isNil(response.data)) {
+        if (response.data.success) return response.data.userProfile
+      }
+      return false
+    })
+    .catch(err => console.log(err))
+}
+
```
Comment (natalieloke) [inline src/services/admin/index.js:175]: Screenshot 2021-04-15 at 9 49 55 AM

there's already a getProfile service in services/user which calls the same endpoint but slightly manipulates the response. wondering if the current getProfile in services/user can be used in place here?

File: src/pages/student/dashboard/mentorships/mentorship-contract/index.js [hunk 3]
```diff
@@ -264,17 +264,23 @@ const MentorshipContractView = () => {
...
+                size="large"
+                onClick={() => setShowRefundModal(true)}
+                icon={<DollarCircleOutlined className="text-info" />}
+                disabled={
+                  isNil(mentorshipContract.mentorPassCount) ||
+                  mentorshipContract.mentorPassCount === 0
+                }
+              >
+                Refund All Passes
+              </Button>
```
Comment (natalieloke) [inline src/pages/student/dashboard/mentorships/mentorship-contract/index.js:281]: is it intended that the button will never be disabled? cos it's in a component which is only rendered when mentorshipContract.mentorPassCount is truthy AND mentorshipContract.mentorPassCount is not 0 (line 267)

i.e both conditions for disabled will never be fulfilled

**Review by natalieloke — COMMENTED** [review]
generally looks ok, thanks @rawrrencee ! left some comments for clarification

File: src/router.js (hunk 1 above)
Comment (rawrrencee) [reply src/router.js:196]: ya lo, oki will do~ needed to refactor other page coz pages/courses alr has the shopping for course pages inside (pages/courses/, pages/courses/view) so i didn't wanna refactor in this PR

File: src/services/admin/index.js (hunk 2 above)
Comment (rawrrencee) [reply src/services/admin/index.js:175]: i need to test this fn but i think can use

File: src/pages/student/dashboard/mentorships/mentorship-contract/index.js (hunk 3 above)
Comment (rawrrencee) [reply src/pages/student/dashboard/mentorships/mentorship-contract/index.js:281]: oo ya double condition, didn't think properly. will remove the disabled condition

**Pull request merged**